application-admintools-default/src/main/java/com/xwiki/admintools/internal/uploadJob/UploadJobFileProcessor.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.uploadJob;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.zip.ZipInputStream;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.AttachmentReference;
+import org.xwiki.model.reference.AttachmentReferenceResolver;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiAttachment;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xwiki.admintools.jobs.JobResult;
+import com.xwiki.admintools.jobs.JobResultLevel;
+import com.xwiki.admintools.jobs.PackageUploadJobStatus;
+import com.xwiki.admintools.uploadPackageJob.UploadPackageJobResource;
+
+/**
+ * Processes the {@link XWikiAttachment} and the given archive file content, while also creating a backup if needed.
+ *
+ * @version $Id$
+ * @since 1.1
+ */
+@Component(roles = UploadJobFileProcessor.class)
+@Singleton
+@Unstable
+public class UploadJobFileProcessor
+{
+    /**
+     * Path format for new files from a given parent.
+     */
+    public static final String NEW_FILE_PATH_FORMAT = "%s/%s";
```

Comment:
Shouldnt hard code path formats, since they might not behave as you expect on all platforms.